tools/simulator/frameworks/runtime-src/proj.ios_mac/mac/SimulatorApp.mm

```diff
@@ -381,27 +380,28 @@ - (void) createWindowAndGLView
     if (pos.x < 0) pos.x = 0;
     if (pos.y < 0) pos.y = 0;
 
-    // create opengl view
-    const cocos2d::Rect frameRect = cocos2d::Rect(0, 0, frameSize.width, frameSize.height);
+    // get app name
     std::stringstream title;
     title << "Cocos Simulator (" << _project.getFrameScale() * 100 << "%)";
-    GLViewImpl *eglView = GLViewImpl::createWithRect(title.str(), frameRect, frameScale, true);
 
-    auto director = Director::getInstance();
-    director->setOpenGLView(eglView);
+    // create opengl view, and init app
+    _app = new AppDelegate(title.str(), _project.getFrameScale() * frameSize.width, _project.getFrameScale() * frameSize.height);
+
+    // this **MUST** be called after create opengl view, or crash will occur at 'gatherGPUInfo'
+    CCLOG("%s\n",Configuration::getInstance()->getInfo().c_str());
 
-    _window = eglView->getCocoaWindow();
-    [[NSApplication sharedApplication] setDelegate: self];
+    auto glfwWindow = ((GLView*)_app->getView())->getGLFWWindow();
+    _window = glfwGetCocoaWindow((GLFWwindow*)glfwWindow);
     [_window center];
 
-    [self setZoom:_project.getFrameScale()];
+    // [self setZoom:_project.getFrameScale()];
     if (pos.x != 0 && pos.y != 0)
     {
         [_window setFrameOrigin:NSMakePoint(pos.x, pos.y)];
     }
 
 #if (PLAYER_SUPPORT_DROP > 0)
-    glfwSetDropCallback(eglView->getWindow(), glfwDropFunc);
+    glfwSetDropCallback((GLFWwindow*)glfwWindow, glfwDropFunc);
```

Comment:
`glfwSetDropCallback ` is used to monitor drop event, for example drop a file into cocos simulator window.
> This function sets the file drop callback of the specified window, which is called when one or more dragged files are dropped on the window.

this feature wasn't used, even cocos creator 1.9.x, might need to be removed later. and much more code have no related with simulator.

Reply:
Sounds reasonable, it is better to add TODO in the codes to avoid forgetting it.